src/kademlia/node.cpp

```diff
@@ -101,22 +102,23 @@ void incoming_error(entry& e, char const* msg, int error_code = 203)
 
 } // anonymous namespace
 
-node::node(udp proto, udp_socket_interface* sock
+node::node(aux::session_listen_socket* sock, socket_manager* sock_man
```

Comment:
you had a consistency argument for using a pointer here instead of a reference, right?
(I think eventually these should be turned into references)